### ISSUE TITLE
OSI LogicalLanes

### ISSUE THREAD

**Issue by ClemensLinnhoff** [esmini/esmini#531]
Are there any plans to implement support for [OSI LogicalLanes](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/V3.6.0/gen/structosi3_1_1LogicalLane.html) in the OSIReporter?

LogicalLanes in OSI are the direct counterpart of OpenDRIVE lanes:
> If OSI is generated from OpenDRIVE, then [LogicalLanes](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/V3.6.0/gen/structosi3_1_1LogicalLane.html) map directly to OpenDRIVE lanes.

How are OSI Lanes currently created in the Reporter? They are also parsed from OpenDrive. Does is make sense to make a distinction between Lanes and LogicalLanes here?

**Comment by eknabevcc** [comment]
I think that in intersections the logical lanes would be helpful since the [intersection type lane](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/V3.7.0/gen/structosi3_1_1Lane_1_1Classification.html) are just mapping incoming and outgoing lanes, not the actual path between them through the intersection. The logical lanes would add those missing paths. 

@ClemensLinnhoff Is this according to your understanding as well? Is this the main benefit of logical lanes, or you see other useful aspects?

**Comment by eknabevcc** [comment]
Looked into it a bit more. 

### Overlapping info
This info not being available already in esmini will require some work - in the end affecting prio. Is this info critical for common use cases or can be skipped until need arise? 

### LogicalLaneBoundary
I first thought logical lanes would only include a reference line and width info - like in OpenDRIVE. Now I understand that the logical boundaries should be included as well, in addition to the reference line. Even considering that some boundaries will be shared between adjacent lanes, it will generate a lot of data. 

Is there a mechanism in OSI to dynamically/on-the-fly choose what classes to populate?

**Comment by eknabevcc** [comment]
Thanks for input, good to know. Let's keep this issue open for a while, inviting for further input.

**Comment by jdsika** [comment]
yes, I think it woud be beneficial to integrate OSMP TrafficParticipant e.g. the [The Stochastic Cognitive Driver Model](https://www.bmwgroup.com/scm-driver/en.html). They are seeking their path and would use the additional information to navigate through the intersections.

**Comment by MichaelSchwarzbachBMW** [comment]
> I see. Good to know. Do you, or someone else, happen to know if also the additional overlapping lane info, e.g. [overlapping_lane](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/V3.7.0/gen/structosi3_1_1LogicalLane.html#sec_12), also is considered by that driver model?

no, the overlapping lane info is currently not considered. This might become an issue when focusing on urban scenarios, but for now this was no issue.

**Comment by eknabevcc** [comment]
Thanks for clarification! Then it could make sense to implement in two steps.

**Comment by jdsika** [comment]
FYI: We want to use the Credible Modeling Process to define the Integration Requirements formally. That maybe useful for you as well to know what to plan next if such TrafficParticipant Models should be integratable in esmini: https://github.com/modelica/ssp-ls-traceability/
